.ci/release/generate_build_matrix.py

```diff
@@ -0,0 +1,520 @@
+        "nvidia-curand-cu12==10.3.9.55; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-cusolver-cu12==11.7.2.55; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-cusparse-cu12==12.5.7.53; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-cusparselt-cu12==0.6.3; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-nccl-cu12==2.26.2; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-nvtx-cu12==12.8.55; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-nvjitlink-cu12==12.8.61; platform_system == 'Linux' and platform_machine == 'x86_64' | "
+        "nvidia-cufile-cu12==1.13.0.11; platform_system == 'Linux' and platform_machine == 'x86_64'"
+    ],
+    # XPU
+    ("xpu", ""): [
+        "intel-cmplr-lib-rt==2025.0.4; platform_system == 'Linux' | "
+        "intel-cmplr-lib-ur==2025.0.4; platform_system == 'Linux' | "
+        "intel-cmplr-lic-rt==2025.0.4; platform_system == 'Linux' | "
+        "intel-sycl-rt==2025.0.4; platform_system == 'Linux' | "
+        "intel-cmplr-lib-rt==2025.0.5; platform_system == 'Windows' | "
+        "intel-cmplr-lib-ur==2025.0.5; platform_system == 'Windows' | "
+        "intel-cmplr-lic-rt==2025.0.5; platform_system == 'Windows' | "
+        "intel-sycl-rt==2025.0.5; platform_system == 'Windows' | "
+        "tcmlib==1.2.0 | "
+        "umf==0.9.1 | "
+        "intel-pti==0.10.1"
+    ],
+}
+
+
+@dataclass
+class BinaryBuild:
+    """
+    Base class representing a PyTorch binary build configuration.
+
+    This class contains common attributes and methods used across different types of builds
+    (CPU, CUDA, ROCm). It defines the build environment, testing infrastructure, and container
+    image specifications.
+
+    Attributes:
+        operating_system (OperatingSystem): Operating system for this build
+        cpu_arch (CpuArch): CPU architecture (x86_64, aarch64, or s390x)
+        accelerator_version (str): Version of the accelerator (e.g., CUDA/ROCm version)
+        accelerator_type (str): Type of accelerator (cpu, cuda, or rocm)
+        builds_on (str): Github Actions runner for building
+        tests_on (str): Github Actions runner for testing
+        arch_list (list[str]): List of specific architectures to build for
+    """
+
+    operating_system: OperatingSystem
+    cpu_arch: CpuArch
+    accelerator_version: str
+    accelerator_type: str
+    builds_on: str = field(default="linux.12xlarge.memory.ephemeral")
+    tests_on: str = field(default="linux.4xlarge")
+    arch_list: list[str] = field(default_factory=list)
+
+    def container_image(self) -> str:
+        """
+        Returns the container image for the build.
+
+        Format: pytorch/wheel-build:{cpu_arch}-{accelerator_type}{accelerator_version}-{DEFAULT_TAG}
```

Comment:
Perhaps to make it easier to manage we should organize as:
``pytorch/wheel-build-{cpu_arch}:{accelerator_type}{accelerator_version}-{DEFAULT_TAG}``
Typically we don't introduce new accelerator types on regular basis, but more 1-2 times a year. And I believe it will probably help keeping arches separate.

Also we need to add on Manylinux flavor in the mix somehow.

Reply:
q: how would this make it easier to manage?

Reply:
Typically different teams work on different accelerator types. Hence having cpu_arch in the name as part of wheel creates natural separation. Basically :
```
pytorch/wheel-build-cuda
pytorch/wheel-build-rocm
pytorch/wheel-build-s390
etc...
```